sympy/physics/vector/functions.py

```diff
@@ -638,3 +639,81 @@ def dynamicsymbols(names, level=0,**assumptions):
 
 dynamicsymbols._t = Symbol('t')  # type: ignore
 dynamicsymbols._str = '\''  # type: ignore
+
+
+def solve_vector(f1, f2, *symbols):
+    """
+    Solves for unknown variables while equating two Vector expressions.
+
+    Parameters
+    ==========
+
+    f1 : Vector1
+
+    f2 : Vector2
+
+    symbols: (object(s) to solve for) specified as
+        - none given(other non-numeric objects will be used)
+        - single symbol
+        - multiple symbols separated by commas
+
+    Examples
+    ========
+
+    The output varies according to the input and can be seen by example:
+
+        >>> from sympy import symbols
+        >>> from sympy.physics.vector import ReferenceFrame
+        >>> from sympy.physics.vector.functions import solve_vector
+        >>> R = ReferenceFrame('R')
+        >>> a, b, c = symbols("a b c", real = True)
+
+        >>> solve_vector(a*R.x + 4*R.y, 6*R.x + b*R.y)
+        {a: 6, b: 4}
+        >>> solve_vector(a*R.x + 4*R.y, 6*R.x + b*R.y,a)
+        {a: 6}
+        >>> solve_vector((a +b +c)*R.x + 2*b*R.y + (a+2*b)*R.z, 6*R.x + 4*R.y +6*R.z)
+        {a: 2, b: 2, c: 2}
+        >>> solve_vector((a**2 + b)*R.x + b*R.y, 6*R.x + 5*R.y)
+        [{a: -1, b: 5}, {a: 1, b: 5}]
+        >>> solve_vector((a**2 + b)*R.x + b*R.y, 5*R.x + 6*R.y)
```

Comment:
Have you tried vector expressions made up of unit vectors from multiple reference frames?

Reply:
I agree with @moorepants, the vector expressions used in examples and tests are trivial. You should check for complex vector expressions also.

Reply:
@sidhu1012 Could you give an example for complex vector expressions ? 
I'm working /correcting another pr of mine and will address comments on this one soon .Thanks for your suggestions !